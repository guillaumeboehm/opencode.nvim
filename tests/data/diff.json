[
  {
    "type": "server.connected",
    "properties": {}
  },
  {
    "type": "session.updated",
    "properties": {
      "info": {
        "id": "ses_6290e7826ffeR7mF2FTlXM52x8",
        "directory": "/Users/cam/tmp/a",
        "time": {
          "updated": 1760247777241,
          "created": 1760247777241
        },
        "version": "0.15.0",
        "title": "junx",
        "projectID": "b0b749d27ca2e03482d36bfe846b01ce40ba759b"
      }
    }
  },
  {
    "type": "session.updated",
    "properties": {
      "info": {
        "id": "ses_628d8425dffe9s3gzVjQ6L1iC2",
        "directory": "/Users/cam/tmp/a",
        "time": {
          "updated": 1760251329954,
          "created": 1760251329954
        },
        "version": "0.15.0",
        "title": "diff",
        "projectID": "b0b749d27ca2e03482d36bfe846b01ce40ba759b"
      }
    }
  },
  {
    "type": "message.updated",
    "properties": {
      "info": {
        "id": "msg_9d7287269001C5gRusYfX7A1w1",
        "sessionID": "ses_628d8425dffe9s3gzVjQ6L1iC2",
        "time": {
          "created": 1760251376233
        },
        "role": "user"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "id": "prt_9d7287269002IBfYpUg2ojuT0E",
        "text": "can you add \"great\" before \"string\" in @diff-test.txt?",
        "sessionID": "ses_628d8425dffe9s3gzVjQ6L1iC2",
        "type": "text",
        "messageID": "msg_9d7287269001C5gRusYfX7A1w1"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "id": "prt_9d728726b001F3dGM6LIJIvf64",
        "text": "Called the Read tool with the following input: {\"filePath\":\"/Users/cam/tmp/a/diff-test.txt\"}",
        "type": "text",
        "sessionID": "ses_628d8425dffe9s3gzVjQ6L1iC2",
        "synthetic": true,
        "messageID": "msg_9d7287269001C5gRusYfX7A1w1"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "id": "prt_9d728726b0020VBtHcq77WPcXF",
        "text": "<file>\n00001| this is a string\n00002| \n</file>",
        "type": "text",
        "sessionID": "ses_628d8425dffe9s3gzVjQ6L1iC2",
        "synthetic": true,
        "messageID": "msg_9d7287269001C5gRusYfX7A1w1"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "id": "prt_9d728726b0032OSbXfc6vhbnnM",
        "source": {
          "type": "file",
          "text": {
            "start": 0,
            "value": "@diff-test.txt",
            "end": 13
          },
          "path": "/Users/cam/tmp/a/diff-test.txt"
        },
        "mime": "text/plain",
        "filename": "diff-test.txt",
        "url": "file:///Users/cam/tmp/a/diff-test.txt",
        "sessionID": "ses_628d8425dffe9s3gzVjQ6L1iC2",
        "type": "file",
        "messageID": "msg_9d7287269001C5gRusYfX7A1w1"
      }
    }
  },
  {
    "type": "session.updated",
    "properties": {
      "info": {
        "id": "ses_628d8425dffe9s3gzVjQ6L1iC2",
        "directory": "/Users/cam/tmp/a",
        "time": {
          "updated": 1760251376239,
          "created": 1760251329954
        },
        "version": "0.15.0",
        "title": "diff",
        "projectID": "b0b749d27ca2e03482d36bfe846b01ce40ba759b"
      }
    }
  },
  {
    "type": "message.updated",
    "properties": {
      "info": {
        "role": "assistant",
        "time": {
          "created": 1760251376263
        },
        "sessionID": "ses_628d8425dffe9s3gzVjQ6L1iC2",
        "id": "msg_9d7287287001HVwpPaH7WkRVdN",
        "tokens": {
          "input": 0,
          "output": 0,
          "cache": {
            "read": 0,
            "write": 0
          },
          "reasoning": 0
        },
        "path": {
          "cwd": "/Users/cam/tmp/a",
          "root": "/Users/cam/tmp/a"
        },
        "mode": "build",
        "providerID": "github-copilot",
        "modelID": "claude-sonnet-4.5",
        "cost": 0
      }
    }
  },
  {
    "type": "session.updated",
    "properties": {
      "info": {
        "id": "ses_628d8425dffe9s3gzVjQ6L1iC2",
        "directory": "/Users/cam/tmp/a",
        "time": {
          "updated": 1760251377970,
          "created": 1760251329954
        },
        "version": "0.15.0",
        "title": "Adding \"great\" before \"string\" in diff-test.txt",
        "projectID": "b0b749d27ca2e03482d36bfe846b01ce40ba759b"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "id": "prt_9d7287aa5001IjzK0nd69jQml7",
        "sessionID": "ses_628d8425dffe9s3gzVjQ6L1iC2",
        "type": "step-start",
        "messageID": "msg_9d7287287001HVwpPaH7WkRVdN"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "id": "prt_9d7287b0d001ZjVVLq0YmMZ6bj",
        "type": "tool",
        "callID": "toolu_vrtx_012A98voL88N5wXanHJhThqS",
        "tool": "edit",
        "sessionID": "ses_628d8425dffe9s3gzVjQ6L1iC2",
        "state": {
          "status": "pending"
        },
        "messageID": "msg_9d7287287001HVwpPaH7WkRVdN"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "id": "prt_9d7287b0d001ZjVVLq0YmMZ6bj",
        "type": "tool",
        "callID": "toolu_vrtx_012A98voL88N5wXanHJhThqS",
        "tool": "edit",
        "sessionID": "ses_628d8425dffe9s3gzVjQ6L1iC2",
        "state": {
          "status": "running",
          "time": {
            "start": 1760251378790
          },
          "input": {
            "oldString": "this is a string",
            "filePath": "/Users/cam/tmp/a/diff-test.txt",
            "newString": "this is a great string"
          }
        },
        "messageID": "msg_9d7287287001HVwpPaH7WkRVdN"
      }
    }
  },
  {
    "type": "permission.updated",
    "properties": {
      "id": "per_9d7287c6c001tkUInBzxaYyRgX",
      "title": "Edit this file: /Users/cam/tmp/a/diff-test.txt",
      "time": {
        "created": 1760251378796
      },
      "callID": "toolu_vrtx_012A98voL88N5wXanHJhThqS",
      "metadata": {
        "filePath": "/Users/cam/tmp/a/diff-test.txt",
        "diff": "Index: /Users/cam/tmp/a/diff-test.txt\n===================================================================\n--- /Users/cam/tmp/a/diff-test.txt\n+++ /Users/cam/tmp/a/diff-test.txt\n@@ -1,1 +1,1 @@\n-this is a string\n+this is a great string\n"
      },
      "sessionID": "ses_628d8425dffe9s3gzVjQ6L1iC2",
      "type": "edit",
      "messageID": "msg_9d7287287001HVwpPaH7WkRVdN"
    }
  },
  {
    "type": "permission.replied",
    "properties": {
      "permissionID": "per_9d7287c6c001tkUInBzxaYyRgX",
      "sessionID": "ses_628d8425dffe9s3gzVjQ6L1iC2",
      "response": "once"
    }
  },
  {
    "type": "file.edited",
    "properties": {
      "file": "/Users/cam/tmp/a/diff-test.txt"
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "id": "prt_9d7287b0d001ZjVVLq0YmMZ6bj",
        "type": "tool",
        "callID": "toolu_vrtx_012A98voL88N5wXanHJhThqS",
        "tool": "edit",
        "sessionID": "ses_628d8425dffe9s3gzVjQ6L1iC2",
        "state": {
          "title": "diff-test.txt",
          "metadata": {
            "diff": "Index: /Users/cam/tmp/a/diff-test.txt\n===================================================================\n--- /Users/cam/tmp/a/diff-test.txt\n+++ /Users/cam/tmp/a/diff-test.txt\n@@ -1,1 +1,1 @@\n-this is a string\n+this is a great string\n",
            "diagnostics": {}
          },
          "status": "completed",
          "output": "",
          "time": {
            "start": 1760251378790,
            "end": 1760251383566
          },
          "input": {
            "oldString": "this is a string",
            "filePath": "/Users/cam/tmp/a/diff-test.txt",
            "newString": "this is a great string"
          }
        },
        "messageID": "msg_9d7287287001HVwpPaH7WkRVdN"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "id": "prt_9d7288f0f001SsnqjRI27JMvcq",
        "tokens": {
          "input": 11471,
          "output": 108,
          "cache": {
            "read": 0,
            "write": 0
          },
          "reasoning": 0
        },
        "cost": 0,
        "sessionID": "ses_628d8425dffe9s3gzVjQ6L1iC2",
        "type": "step-finish",
        "messageID": "msg_9d7287287001HVwpPaH7WkRVdN"
      }
    }
  },
  {
    "type": "message.updated",
    "properties": {
      "info": {
        "role": "assistant",
        "time": {
          "created": 1760251376263
        },
        "sessionID": "ses_628d8425dffe9s3gzVjQ6L1iC2",
        "id": "msg_9d7287287001HVwpPaH7WkRVdN",
        "tokens": {
          "input": 11471,
          "output": 108,
          "cache": {
            "read": 0,
            "write": 0
          },
          "reasoning": 0
        },
        "path": {
          "cwd": "/Users/cam/tmp/a",
          "root": "/Users/cam/tmp/a"
        },
        "mode": "build",
        "providerID": "github-copilot",
        "modelID": "claude-sonnet-4.5",
        "cost": 0
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "id": "prt_9d7288f2e001IzI7WzwdrQhZ5p",
        "hash": "1f593f7ed419c95d3995f8ef4b98d4e571c3a492",
        "files": ["/Users/cam/tmp/a/diff-test.txt"],
        "sessionID": "ses_628d8425dffe9s3gzVjQ6L1iC2",
        "type": "patch",
        "messageID": "msg_9d7287287001HVwpPaH7WkRVdN"
      }
    }
  },
  {
    "type": "message.updated",
    "properties": {
      "info": {
        "role": "assistant",
        "time": {
          "completed": 1760251383598,
          "created": 1760251376263
        },
        "sessionID": "ses_628d8425dffe9s3gzVjQ6L1iC2",
        "id": "msg_9d7287287001HVwpPaH7WkRVdN",
        "tokens": {
          "input": 11471,
          "output": 108,
          "cache": {
            "read": 0,
            "write": 0
          },
          "reasoning": 0
        },
        "path": {
          "cwd": "/Users/cam/tmp/a",
          "root": "/Users/cam/tmp/a"
        },
        "mode": "build",
        "providerID": "github-copilot",
        "modelID": "claude-sonnet-4.5",
        "cost": 0
      }
    }
  },
  {
    "type": "message.updated",
    "properties": {
      "info": {
        "role": "assistant",
        "time": {
          "completed": 1760251383599,
          "created": 1760251376263
        },
        "sessionID": "ses_628d8425dffe9s3gzVjQ6L1iC2",
        "id": "msg_9d7287287001HVwpPaH7WkRVdN",
        "tokens": {
          "input": 11471,
          "output": 108,
          "cache": {
            "read": 0,
            "write": 0
          },
          "reasoning": 0
        },
        "path": {
          "cwd": "/Users/cam/tmp/a",
          "root": "/Users/cam/tmp/a"
        },
        "mode": "build",
        "providerID": "github-copilot",
        "modelID": "claude-sonnet-4.5",
        "cost": 0
      }
    }
  },
  {
    "type": "message.updated",
    "properties": {
      "info": {
        "role": "assistant",
        "time": {
          "completed": 1760251383599,
          "created": 1760251376263
        },
        "sessionID": "ses_628d8425dffe9s3gzVjQ6L1iC2",
        "id": "msg_9d7287287001HVwpPaH7WkRVdN",
        "tokens": {
          "input": 11471,
          "output": 108,
          "cache": {
            "read": 0,
            "write": 0
          },
          "reasoning": 0
        },
        "path": {
          "cwd": "/Users/cam/tmp/a",
          "root": "/Users/cam/tmp/a"
        },
        "mode": "build",
        "providerID": "github-copilot",
        "modelID": "claude-sonnet-4.5",
        "cost": 0
      }
    }
  },
  {
    "type": "message.updated",
    "properties": {
      "info": {
        "role": "assistant",
        "time": {
          "created": 1760251383599
        },
        "sessionID": "ses_628d8425dffe9s3gzVjQ6L1iC2",
        "id": "msg_9d7288f2f001hW6NqqhtBc72UU",
        "tokens": {
          "input": 0,
          "output": 0,
          "cache": {
            "read": 0,
            "write": 0
          },
          "reasoning": 0
        },
        "path": {
          "cwd": "/Users/cam/tmp/a",
          "root": "/Users/cam/tmp/a"
        },
        "mode": "build",
        "providerID": "github-copilot",
        "modelID": "claude-sonnet-4.5",
        "cost": 0
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "id": "prt_9d728945d001x9E3ruZp9fMsQs",
        "sessionID": "ses_628d8425dffe9s3gzVjQ6L1iC2",
        "type": "step-start",
        "messageID": "msg_9d7288f2f001hW6NqqhtBc72UU"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "id": "prt_9d72894a100187xxUJg2VLqqGX",
        "tokens": {
          "input": 11606,
          "output": 2,
          "cache": {
            "read": 11081,
            "write": 0
          },
          "reasoning": 0
        },
        "cost": 0,
        "sessionID": "ses_628d8425dffe9s3gzVjQ6L1iC2",
        "type": "step-finish",
        "messageID": "msg_9d7288f2f001hW6NqqhtBc72UU"
      }
    }
  },
  {
    "type": "message.updated",
    "properties": {
      "info": {
        "role": "assistant",
        "time": {
          "created": 1760251383599
        },
        "sessionID": "ses_628d8425dffe9s3gzVjQ6L1iC2",
        "id": "msg_9d7288f2f001hW6NqqhtBc72UU",
        "tokens": {
          "input": 11606,
          "output": 2,
          "cache": {
            "read": 11081,
            "write": 0
          },
          "reasoning": 0
        },
        "path": {
          "cwd": "/Users/cam/tmp/a",
          "root": "/Users/cam/tmp/a"
        },
        "mode": "build",
        "providerID": "github-copilot",
        "modelID": "claude-sonnet-4.5",
        "cost": 0
      }
    }
  },
  {
    "type": "message.updated",
    "properties": {
      "info": {
        "role": "assistant",
        "time": {
          "completed": 1760251385040,
          "created": 1760251383599
        },
        "sessionID": "ses_628d8425dffe9s3gzVjQ6L1iC2",
        "id": "msg_9d7288f2f001hW6NqqhtBc72UU",
        "tokens": {
          "input": 11606,
          "output": 2,
          "cache": {
            "read": 11081,
            "write": 0
          },
          "reasoning": 0
        },
        "path": {
          "cwd": "/Users/cam/tmp/a",
          "root": "/Users/cam/tmp/a"
        },
        "mode": "build",
        "providerID": "github-copilot",
        "modelID": "claude-sonnet-4.5",
        "cost": 0
      }
    }
  },
  {
    "type": "message.updated",
    "properties": {
      "info": {
        "role": "assistant",
        "time": {
          "completed": 1760251385041,
          "created": 1760251383599
        },
        "sessionID": "ses_628d8425dffe9s3gzVjQ6L1iC2",
        "id": "msg_9d7288f2f001hW6NqqhtBc72UU",
        "tokens": {
          "input": 11606,
          "output": 2,
          "cache": {
            "read": 11081,
            "write": 0
          },
          "reasoning": 0
        },
        "path": {
          "cwd": "/Users/cam/tmp/a",
          "root": "/Users/cam/tmp/a"
        },
        "mode": "build",
        "providerID": "github-copilot",
        "modelID": "claude-sonnet-4.5",
        "cost": 0
      }
    }
  },
  {
    "type": "message.updated",
    "properties": {
      "info": {
        "role": "assistant",
        "time": {
          "completed": 1760251385041,
          "created": 1760251383599
        },
        "sessionID": "ses_628d8425dffe9s3gzVjQ6L1iC2",
        "id": "msg_9d7288f2f001hW6NqqhtBc72UU",
        "tokens": {
          "input": 11606,
          "output": 2,
          "cache": {
            "read": 11081,
            "write": 0
          },
          "reasoning": 0
        },
        "path": {
          "cwd": "/Users/cam/tmp/a",
          "root": "/Users/cam/tmp/a"
        },
        "mode": "build",
        "providerID": "github-copilot",
        "modelID": "claude-sonnet-4.5",
        "cost": 0
      }
    }
  },
  {
    "type": "session.idle",
    "properties": {
      "sessionID": "ses_628d8425dffe9s3gzVjQ6L1iC2"
    }
  }
]
