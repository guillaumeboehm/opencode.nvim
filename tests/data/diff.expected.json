{"lines":["","----","","","can you add \"great\" before \"string\" in @diff-test.txt?","","[diff-test.txt](diff-test.txt)","","----","","","**  edit** `diff-test.txt`","","```txt"," this is a string"," this is a great string","","```","","**󰻛  Created Snapshot** `1f593f7e`","","----","",""],"timestamp":1761108013,"actions":[{"type":"diff_revert_selected_file","key":"R","text":"[R]evert file","args":["1f593f7ed419c95d3995f8ef4b98d4e571c3a492"],"range":{"from":20,"to":20},"display_line":20},{"type":"diff_revert_all","key":"A","text":"Revert [A]ll","args":["1f593f7ed419c95d3995f8ef4b98d4e571c3a492"],"range":{"from":20,"to":20},"display_line":20},{"type":"diff_open","key":"D","text":"[D]iff","args":["1f593f7ed419c95d3995f8ef4b98d4e571c3a492"],"range":{"from":20,"to":20},"display_line":20}],"extmarks":[[1,2,0,{"virt_text_hide":false,"virt_text_pos":"win_col","virt_text_repeat_linebreak":false,"virt_text":[["▌󰭻 ","OpencodeMessageRoleUser"],[" "],["USER","OpencodeMessageRoleUser"],["","OpencodeHint"],[" (2025-10-12 06:42:56)","OpencodeHint"],[" [msg_9d7287269001C5gRusYfX7A1w1]","OpencodeHint"]],"priority":10,"right_gravity":true,"ns_id":3,"virt_text_win_col":-3}],[2,3,0,{"virt_text_hide":false,"virt_text_pos":"win_col","virt_text_repeat_linebreak":true,"virt_text":[["▌","OpencodeMessageRoleUser"]],"priority":4096,"right_gravity":true,"ns_id":3,"virt_text_win_col":-3}],[3,4,0,{"virt_text_hide":false,"virt_text_pos":"win_col","virt_text_repeat_linebreak":true,"virt_text":[["▌","OpencodeMessageRoleUser"]],"priority":4096,"right_gravity":true,"ns_id":3,"virt_text_win_col":-3}],[4,4,39,{"end_row":4,"ns_id":3,"hl_group":"OpencodeMention","end_col":53,"priority":1000,"hl_eol":false,"right_gravity":true,"end_right_gravity":false}],[5,5,0,{"virt_text_hide":false,"virt_text_pos":"win_col","virt_text_repeat_linebreak":true,"virt_text":[["▌","OpencodeMessageRoleUser"]],"priority":4096,"right_gravity":true,"ns_id":3,"virt_text_win_col":-3}],[6,6,0,{"virt_text_hide":false,"virt_text_pos":"win_col","virt_text_repeat_linebreak":true,"virt_text":[["▌","OpencodeMessageRoleUser"]],"priority":4096,"right_gravity":true,"ns_id":3,"virt_text_win_col":-3}],[7,9,0,{"virt_text_hide":false,"virt_text_pos":"win_col","virt_text_repeat_linebreak":false,"virt_text":[[" ","OpencodeMessageRoleAssistant"],[" "],["BUILD","OpencodeMessageRoleAssistant"],[" claude-sonnet-4.5","OpencodeHint"],[" (2025-10-12 06:42:56)","OpencodeHint"],[" [msg_9d7287287001HVwpPaH7WkRVdN]","OpencodeHint"]],"priority":10,"right_gravity":true,"ns_id":3,"virt_text_win_col":-3}],[8,11,0,{"virt_text_hide":false,"virt_text_pos":"win_col","virt_text_repeat_linebreak":true,"virt_text":[["▌","OpencodeToolBorder"]],"priority":4096,"right_gravity":true,"ns_id":3,"virt_text_win_col":-1}],[9,12,0,{"virt_text_hide":false,"virt_text_pos":"win_col","virt_text_repeat_linebreak":true,"virt_text":[["▌","OpencodeToolBorder"]],"priority":4096,"right_gravity":true,"ns_id":3,"virt_text_win_col":-1}],[10,13,0,{"virt_text_hide":false,"virt_text_pos":"win_col","virt_text_repeat_linebreak":true,"virt_text":[["▌","OpencodeToolBorder"]],"priority":4096,"right_gravity":true,"ns_id":3,"virt_text_win_col":-1}],[11,14,0,{"end_col":0,"end_row":15,"hl_group":"OpencodeDiffDelete","right_gravity":true,"end_right_gravity":false,"virt_text_hide":false,"virt_text_pos":"overlay","virt_text_repeat_linebreak":false,"virt_text":[["-","OpencodeDiffDelete"]],"priority":5000,"ns_id":3,"hl_eol":true}],[12,14,0,{"virt_text_hide":false,"virt_text_pos":"win_col","virt_text_repeat_linebreak":true,"virt_text":[["▌","OpencodeToolBorder"]],"priority":4096,"right_gravity":true,"ns_id":3,"virt_text_win_col":-1}],[13,15,0,{"end_col":0,"end_row":16,"hl_group":"OpencodeDiffAdd","right_gravity":true,"end_right_gravity":false,"virt_text_hide":false,"virt_text_pos":"overlay","virt_text_repeat_linebreak":false,"virt_text":[["+","OpencodeDiffAdd"]],"priority":5000,"ns_id":3,"hl_eol":true}],[14,15,0,{"virt_text_hide":false,"virt_text_pos":"win_col","virt_text_repeat_linebreak":true,"virt_text":[["▌","OpencodeToolBorder"]],"priority":4096,"right_gravity":true,"ns_id":3,"virt_text_win_col":-1}],[15,16,0,{"virt_text_hide":false,"virt_text_pos":"win_col","virt_text_repeat_linebreak":true,"virt_text":[["▌","OpencodeToolBorder"]],"priority":4096,"right_gravity":true,"ns_id":3,"virt_text_win_col":-1}],[16,17,0,{"virt_text_hide":false,"virt_text_pos":"win_col","virt_text_repeat_linebreak":true,"virt_text":[["▌","OpencodeToolBorder"]],"priority":4096,"right_gravity":true,"ns_id":3,"virt_text_win_col":-1}],[17,22,0,{"virt_text_hide":false,"virt_text_pos":"win_col","virt_text_repeat_linebreak":false,"virt_text":[[" ","OpencodeMessageRoleAssistant"],[" "],["BUILD","OpencodeMessageRoleAssistant"],[" claude-sonnet-4.5","OpencodeHint"],[" (2025-10-12 06:43:03)","OpencodeHint"],[" [msg_9d7288f2f001hW6NqqhtBc72UU]","OpencodeHint"]],"priority":10,"right_gravity":true,"ns_id":3,"virt_text_win_col":-3}]]}